### ISSUE TITLE
Keyboard shortcut manager plugin

### ISSUE THREAD

**Issue by lorensr** [zeit/hyper#251]
I'm interested in a plugin that allows the user to assign keys to the following:
1. a few included commands like forward-word (moves cursor to end of word) 
2. commands provided by another extensions (for instance [open new tab and cd to same directory](https://github.com/hharnisc/hypercwd/issues/1#issuecomment-233205409))
3. arbitrary JS? (unless this would be ~as difficult as publishing your own extension)

Maybe something like this:

``` js
module.exports = {
  config: {
    keys: [
      'alt-f': 'forward-word',
      'ctrl-alt-t': 'hypercwd/new-tab-current-dir',
      'ctrl-w': (store) -> {
        store.dispatch({
          type: 'WOW_MODE_TOGGLE'
        });
      }
    ]
  }
}
```

Anyone have advice on the best way to architect this? cc @hharnisc

**Comment by hharnisc** [comment]
Starting to think about this. I'm looking at this chunk of code https://github.com/zeit/hyperterm/blob/master/app/lib/containers/hyperterm.js#L31-L64

I'm thinking of making these props that can be overridden and/or extended. Just an idea at this point though.

**Comment by iamstarkov** [comment]
it will not work as a standalone plugin with current code base

**Comment by iamstarkov** [comment]
i made a relevant proposal to discuss #872

**Comment by iamstarkov** [comment]
check #925 out

**Comment by ppot** [comment]
closed in #1876
